client/src/webview-js/image-upload/image-upload.ts

```diff
@@ -1,22 +1,30 @@
-let vscode
-let pending = []
+declare let acquireVsCodeApi: any
+let vscode: any
+
+interface PendingEntry {
+  mediaName: string
+  data: string | ArrayBuffer | null
```

Comment:
We should only expect data to be a `string`
```suggestion
  data: string
```
because we only use `readDataAsUrl()` on the `FileReader` below, it should be fine for us to cast `reader.result` to `string` when we push onto `pending`. e.g.
```js
pending.push({
  mediaName: file.name,
  data: reader.result as string
})
```